.storybook/preview.tsx

```diff
@@ -1,25 +1,20 @@
 import '@atlaskit/css-reset';
 
+import { DecoratorFn } from '@storybook/react';
 import React from 'react';
-import { withPerformance } from 'storybook-addon-performance';
 import { resetData } from '../stories/src/data';
 import GlobalStyles from './custom-decorators/global-styles';
 import welcomeMessage from './welcome-message';
 
 welcomeMessage();
 
-export const decorators = [
-  (Story: React.ElementType, { id }: { id: string }) => {
+export const decorators: DecoratorFn[] = [
+  (story, { id }: { id: string }) => {
     resetData(id);
 
-    return <Story key={id} />;
+    return story({ key: id });
   },
-  (Story: React.ElementType): React.ReactElement => (
-    <GlobalStyles>
-      <Story />
-    </GlobalStyles>
-  ),
-  withPerformance,
+  (story) => <GlobalStyles>{story()}</GlobalStyles>,
 ];
```

Comment:
There was an issue with the strict mode, because of we were using the `story` as a component.

see: https://github.com/storybookjs/storybook/issues/15223#issuecomment-1092837912